lib/domain.js

```diff
@@ -53,8 +53,26 @@ Object.defineProperty(process, 'domain', {
 const stack = [];
 exports._stack = stack;
 
-// let the process know we're using domains
-const _domain_flag = process._setupDomainUse(_domain, stack);
+class Domain extends EventEmitter {
+
+  constructor() {
+    super();
+
+    this.members = [];
+    this.asyncIds = new Set();
+    this.asyncHook = null; // will hold the asyncHook after _enable is called
+    this._enable();
```

Comment:
this will make every domain created register new async_hooks. I'm not 100% sure this is the intended usage, and what side effect it would create. cc @AndreasMadsen

Reply:
Yeah, I think a `Map` that keeps track of the `async id` → `domain` mapping would be a better idea…

Reply:
Should be good!

Reply:
Updated